### PR TITLE
Upgrade terraform-plugin-sdk to v2+

### PR DESCRIPTION
**Pull request by davidji99**
Using the [official documentation](https://www.terraform.io/docs/extend/guides/v2-upgrade-guide.html) as a guide.

Once this issue/PR goes in, a major version bump is needed as terraform `v0.11.X` will no longer be supported.

Also need to bump https://github.com/heroku/terraform-provider-heroku/blob/master/go.mod#L1 likely.

I'm also dropping the `vendor` directory since I don't believe this is needed.

### CODE REVIEW

**Comment by davidji99** [comment]
Failing tests:
```
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./heroku/ -v  -timeout=240m -ldflags="-X=github.com/heroku/terraform-provider-heroku/v4/version.ProviderVersion=test"
=== RUN   TestAccDatasourceHerokuAddon_Basic
--- PASS: TestAccDatasourceHerokuAddon_Basic (13.25s)
=== RUN   TestAccDatasourceHerokuApp_Basic
--- PASS: TestAccDatasourceHerokuApp_Basic (8.02s)
=== RUN   TestAccDatasourceHerokuApp_Advanced
--- PASS: TestAccDatasourceHerokuApp_Advanced (440.11s)
=== RUN   TestAccDatasourceHerokuPipeline_Basic
--- PASS: TestAccDatasourceHerokuPipeline_Basic (9.04s)
=== RUN   TestAccDatasourceHerokuSpacePeeringInfo_Basic
--- PASS: TestAccDatasourceHerokuSpacePeeringInfo_Basic (494.53s)
=== RUN   TestAccDatasourceHerokuSpace_Basic
--- PASS: TestAccDatasourceHerokuSpace_Basic (477.89s)
=== RUN   TestAccDatasourceHerokuTeam_Basic
--- PASS: TestAccDatasourceHerokuTeam_Basic (1.36s)
=== RUN   TestAccHerokuAccountFeature_importBasic
--- PASS: TestAccHerokuAccountFeature_importBasic (1.71s)
=== RUN   TestAccHerokuAddonAttachment_importBasic
--- PASS: TestAccHerokuAddonAttachment_importBasic (8.90s)
=== RUN   TestAccHerokuAddon_importBasic
--- PASS: TestAccHerokuAddon_importBasic (9.69s)
=== RUN   TestAccHerokuAppFeature_importBasic
--- PASS: TestAccHerokuAppFeature_importBasic (7.80s)
=== RUN   TestAccHerokuAppRelease_importBasic
--- PASS: TestAccHerokuAppRelease_importBasic (7.90s)
=== RUN   TestAccHerokuApp_importBasic
--- PASS: TestAccHerokuApp_importBasic (7.93s)
=== RUN   TestAccHerokuApp_importOrganization
--- PASS: TestAccHerokuApp_importOrganization (8.29s)
=== RUN   TestAccHerokuApp_importBuildpacks
--- PASS: TestAccHerokuApp_importBuildpacks (7.89s)
=== RUN   TestAccHerokuAppWebhook_importBasic
--- PASS: TestAccHerokuAppWebhook_importBasic (9.36s)
=== RUN   TestAccHerokuBuild_importBasic
Error:     import_heroku_build_test.go:14: Step 1/2 error: Error running pre-apply refresh: exit status 1
        
        Error: Error stating build source path : stat : no such file or directory
        
          on terraform_plugin_test.tf line 6, in resource "heroku_build" "foobar":
           6: resource "heroku_build" "foobar" {
        
        
--- FAIL: TestAccHerokuBuild_importBasic (0.34s)
=== RUN   TestAccHerokuBuild_importAllOpts
Error:     import_heroku_build_test.go:35: Step 1/2 error: Error running pre-apply refresh: exit status 1
        
        Error: Error stating build source path : stat : no such file or directory
        
          on terraform_plugin_test.tf line 6, in resource "heroku_build" "foobar":
           6: resource "heroku_build" "foobar" {
        
        
--- FAIL: TestAccHerokuBuild_importAllOpts (0.41s)
=== RUN   TestAccHerokuBuild_importWithFileUrl
Error:     import_heroku_build_test.go:56: Step 1/2 error: Error running apply: exit status 1
        
        Error: source.checksum should be empty when source.path is set (checksum is auto-generated)
        
        
--- FAIL: TestAccHerokuBuild_importWithFileUrl (6.28s)
=== RUN   TestAccHerokuCert_importBasic
--- PASS: TestAccHerokuCert_importBasic (21.59s)
=== RUN   TestAccHerokuCollaborator_importBasic
--- PASS: TestAccHerokuCollaborator_importBasic (7.93s)
=== RUN   TestAccHerokuAppConfigAssociation_importBasic
--- PASS: TestAccHerokuAppConfigAssociation_importBasic (8.05s)
=== RUN   TestAccHerokuConfig_importBasic
--- PASS: TestAccHerokuConfig_importBasic (1.23s)
=== RUN   TestAccHerokuDomain_importBasic
--- PASS: TestAccHerokuDomain_importBasic (12.89s)
=== RUN   TestAccHerokuDrain_importBasic
--- PASS: TestAccHerokuDrain_importBasic (7.93s)
=== RUN   TestAccHerokuFormation_importBasic
--- PASS: TestAccHerokuFormation_importBasic (8.22s)
=== RUN   TestAccHerokuPipelineConfigVar_importBasic
--- PASS: TestAccHerokuPipelineConfigVar_importBasic (2.05s)
=== RUN   TestAccHerokuPipelineCoupling_importBasic
--- PASS: TestAccHerokuPipelineCoupling_importBasic (9.35s)
=== RUN   TestAccHerokuPipeline_importBasic
--- PASS: TestAccHerokuPipeline_importBasic (1.87s)
=== RUN   TestAccHerokuSlug_importBasic
--- PASS: TestAccHerokuSlug_importBasic (7.79s)
=== RUN   TestAccHerokuSlug_importAllOpts
--- PASS: TestAccHerokuSlug_importAllOpts (7.68s)
=== RUN   TestAccHerokuSlug_importWithFileUrl
--- PASS: TestAccHerokuSlug_importWithFileUrl (8.10s)
=== RUN   TestAccHerokuSpaceAppAccess_importBasic
--- PASS: TestAccHerokuSpaceAppAccess_importBasic (462.07s)
=== RUN   TestAccHerokuTeamCollaborator_importBasic
--- PASS: TestAccHerokuTeamCollaborator_importBasic (8.36s)
=== RUN   TestAccHerokuTeamMember_importBasic
--- PASS: TestAccHerokuTeamMember_importBasic (2.04s)
=== RUN   TestProvider
--- PASS: TestProvider (0.00s)
=== RUN   TestProvider_impl
--- PASS: TestProvider_impl (0.00s)
=== RUN   TestProviderConfigureUsesHeadersForClient
--- PASS: TestProviderConfigureUsesHeadersForClient (0.00s)
=== RUN   TestAccHerokuAccountFeature_Basic
--- PASS: TestAccHerokuAccountFeature_Basic (1.49s)
=== RUN   TestAccHerokuAddonAttachment_Basic
--- PASS: TestAccHerokuAddonAttachment_Basic (8.47s)
=== RUN   TestAccHerokuAddonAttachment_Named
--- PASS: TestAccHerokuAddonAttachment_Named (8.58s)
=== RUN   TestAccHerokuAddon_Basic
--- PASS: TestAccHerokuAddon_Basic (7.87s)
=== RUN   TestAccHerokuAddon_noPlan
--- PASS: TestAccHerokuAddon_noPlan (10.11s)
=== RUN   TestAccHerokuAddon_CustomName
--- PASS: TestAccHerokuAddon_CustomName (8.53s)
=== RUN   TestAccHerokuAddon_CustomName_Invalid
Error:     resource_heroku_addon_test.go:107: Step 1/1, expected an error with pattern, no match on: Error running pre-apply refresh: exit status 1
        
        Error: invalid value for name (Invalid custom addon name: must start with a letter and can only contain lowercase letters, numbers, and dashes)
        
          on terraform_plugin_test.tf line 10, in resource "heroku_addon" "foobar":
          10:     name = "da.%dsadsa$d"
        
        
--- FAIL: TestAccHerokuAddon_CustomName_Invalid (0.39s)
=== RUN   TestAccHerokuAddon_CustomName_EmptyString
Error:     resource_heroku_addon_test.go:124: Step 1/1, expected an error with pattern, no match on: Error running pre-apply refresh: exit status 1
        
        Error: expected length of name to be in the range (1 - 256), got 
        
          on terraform_plugin_test.tf line 10, in resource "heroku_addon" "foobar":
          10:     name = ""
        
        
        
        Error: invalid value for name (Invalid custom addon name: must start with a letter and can only contain lowercase letters, numbers, and dashes)
        
          on terraform_plugin_test.tf line 10, in resource "heroku_addon" "foobar":
          10:     name = ""
        
        
--- FAIL: TestAccHerokuAddon_CustomName_EmptyString (0.35s)
=== RUN   TestAccHerokuAddon_CustomName_FirstCharNum
Error:     resource_heroku_addon_test.go:141: Step 1/1, expected an error with pattern, no match on: Error running pre-apply refresh: exit status 1
        
        Error: invalid value for name (Invalid custom addon name: must start with a letter and can only contain lowercase letters, numbers, and dashes)
        
          on terraform_plugin_test.tf line 10, in resource "heroku_addon" "foobar":
          10:     name = "1dasdad"
        
        
--- FAIL: TestAccHerokuAddon_CustomName_FirstCharNum (0.36s)
=== RUN   TestAccHerokuAddon_Disappears
--- PASS: TestAccHerokuAddon_Disappears (7.83s)
=== RUN   TestAccHerokuAppConfigAssociation_Basic
--- PASS: TestAccHerokuAppConfigAssociation_Basic (7.88s)
=== RUN   TestAccHerokuAppConfigAssociation_Advanced
--- PASS: TestAccHerokuAppConfigAssociation_Advanced (8.06s)
=== RUN   TestAccHerokuAppFeature
--- PASS: TestAccHerokuAppFeature (9.17s)
=== RUN   TestAccHerokuAppRelease_Basic
--- PASS: TestAccHerokuAppRelease_Basic (7.47s)
=== RUN   TestAccHerokuAppRelease_OrgBasic
--- PASS: TestAccHerokuAppRelease_OrgBasic (7.83s)
=== RUN   TestAccHerokuApp_Basic
--- PASS: TestAccHerokuApp_Basic (7.14s)
=== RUN   TestAccHerokuApp_Disappears
--- PASS: TestAccHerokuApp_Disappears (6.83s)
=== RUN   TestAccHerokuApp_Change
--- PASS: TestAccHerokuApp_Change (9.24s)
=== RUN   TestAccHerokuApp_NukeVars
--- PASS: TestAccHerokuApp_NukeVars (9.11s)
=== RUN   TestAccHerokuApp_Buildpacks
--- PASS: TestAccHerokuApp_Buildpacks (11.16s)
=== RUN   TestAccHerokuApp_ExternallySetBuildpacks
--- PASS: TestAccHerokuApp_ExternallySetBuildpacks (9.21s)
=== RUN   TestAccHerokuApp_ACM
--- PASS: TestAccHerokuApp_ACM (11.62s)
=== RUN   TestAccHerokuApp_Organization
Error:     resource_heroku_app_test.go:256: Step 1/1 error: Check failed: Check 2/2 error: Bad stack: heroku-20
--- FAIL: TestAccHerokuApp_Organization (6.65s)
=== RUN   TestAccHerokuApp_Space
Error:     resource_heroku_app_test.go:280: Step 1/1 error: Check failed: Check 2/2 error: Bad stack: heroku-20
--- FAIL: TestAccHerokuApp_Space (508.21s)
=== RUN   TestAccHerokuApp_Space_Internal
Error:     resource_heroku_app_test.go:304: Step 1/1 error: Check failed: Check 2/2 error: Bad stack: heroku-20
--- FAIL: TestAccHerokuApp_Space_Internal (498.08s)
=== RUN   TestAccHerokuApp_EmptyConfigVars
--- PASS: TestAccHerokuApp_EmptyConfigVars (7.17s)
=== RUN   TestAccHerokuApp_SensitiveConfigVars
--- PASS: TestAccHerokuApp_SensitiveConfigVars (11.52s)
=== RUN   TestAccHerokuApp_Organization_Locked
--- PASS: TestAccHerokuApp_Organization_Locked (9.56s)
=== RUN   TestAccHerokuAppWebhook_Basic
--- PASS: TestAccHerokuAppWebhook_Basic (11.41s)
=== RUN   TestAccHerokuBuild_Basic
Error:     resource_heroku_build_test.go:21: Step 1/1 error: Error running pre-apply refresh: exit status 1
        
        Error: Error stating build source path : stat : no such file or directory
        
          on terraform_plugin_test.tf line 6, in resource "heroku_build" "foobar":
           6: resource "heroku_build" "foobar" {
        
        
--- FAIL: TestAccHerokuBuild_Basic (0.32s)
=== RUN   TestAccHerokuBuild_InsecureUrl
--- PASS: TestAccHerokuBuild_InsecureUrl (0.32s)
=== RUN   TestAccHerokuBuild_NoSource
Error:     resource_heroku_build_test.go:56: Step 1/1, expected an error with pattern, no match on: Error running pre-apply refresh: exit status 1
        
        Error: Error stating build source path : stat : no such file or directory
        
          on terraform_plugin_test.tf line 6, in resource "heroku_build" "foobar":
           6: resource "heroku_build" "foobar" {
        
        
--- FAIL: TestAccHerokuBuild_NoSource (0.33s)
=== RUN   TestAccHerokuBuild_AllOpts
Error:     resource_heroku_build_test.go:73: Step 1/1 error: Error running pre-apply refresh: exit status 1
        
        Error: Error stating build source path : stat : no such file or directory
        
          on terraform_plugin_test.tf line 6, in resource "heroku_build" "foobar":
           6: resource "heroku_build" "foobar" {
        
        
--- FAIL: TestAccHerokuBuild_AllOpts (0.32s)
=== RUN   TestAccHerokuBuild_LocalSourceTarball
Error:     resource_heroku_build_test.go:98: Step 1/2 error: Error running apply: exit status 1
        
        Error: source.checksum should be empty when source.path is set (checksum is auto-generated)
        
        
--- FAIL: TestAccHerokuBuild_LocalSourceTarball (6.37s)
=== RUN   TestAccHerokuBuild_LocalSourceTarball_SetChecksum
--- PASS: TestAccHerokuBuild_LocalSourceTarball_SetChecksum (6.28s)
=== RUN   TestAccHerokuBuild_LocalSourceTarball_AllOpts
Error:     resource_heroku_build_test.go:142: Step 1/1 error: Error running apply: exit status 1
        
        Error: source.checksum should be empty when source.path is set (checksum is auto-generated)
        
        
--- FAIL: TestAccHerokuBuild_LocalSourceTarball_AllOpts (6.36s)
=== RUN   TestAccHerokuBuild_LocalSourceDirectory
Error:     resource_heroku_build_test.go:164: Step 1/2 error: Error running apply: exit status 1
        
        Error: source.checksum should be empty when source.path is set (checksum is auto-generated)
        
        
--- FAIL: TestAccHerokuBuild_LocalSourceDirectory (6.34s)
=== RUN   TestAccHerokuBuild_LocalSourceDirectorySelfContained
Error:     resource_heroku_build_test.go:196: Step 1/1 error: Error running apply: exit status 1
        
        Error: source.checksum should be empty when source.path is set (checksum is auto-generated)
        
        
--- FAIL: TestAccHerokuBuild_LocalSourceDirectorySelfContained (6.32s)
=== RUN   TestAccHerokuCert_EU
--- PASS: TestAccHerokuCert_EU (37.86s)
=== RUN   TestAccHerokuCert_US
--- PASS: TestAccHerokuCert_US (36.78s)
=== RUN   TestAccHerokuCollaborator_Basic
--- PASS: TestAccHerokuCollaborator_Basic (7.64s)
=== RUN   TestAccHerokuConfig_Single
--- PASS: TestAccHerokuConfig_Single (1.09s)
=== RUN   TestAccHerokuConfig_Both
--- PASS: TestAccHerokuConfig_Both (1.14s)
=== RUN   TestAccHerokuConfig_Dupe
--- PASS: TestAccHerokuConfig_Dupe (0.58s)
=== RUN   TestAccHerokuDomain_Basic
--- PASS: TestAccHerokuDomain_Basic (12.45s)
=== RUN   TestAccHerokuDrain_Basic
--- PASS: TestAccHerokuDrain_Basic (7.56s)
=== RUN   TestAccHerokuFormationSingleUpdate_WithOrg
--- PASS: TestAccHerokuFormationSingleUpdate_WithOrg (7.67s)
=== RUN   TestAccHerokuFormationUpdateFreeDyno
--- PASS: TestAccHerokuFormationUpdateFreeDyno (7.44s)
=== RUN   TestAccHerokuPipelineConfigVar_TestStage_Basic
--- PASS: TestAccHerokuPipelineConfigVar_TestStage_Basic (1.74s)
=== RUN   TestAccHerokuPipelineConfigVar_ReviewStage_Basic
--- PASS: TestAccHerokuPipelineConfigVar_ReviewStage_Basic (1.72s)
=== RUN   TestAccHerokuPipelineCoupling_Basic
--- PASS: TestAccHerokuPipelineCoupling_Basic (8.77s)
=== RUN   TestAccHerokuPipeline_Basic
--- PASS: TestAccHerokuPipeline_Basic (2.53s)
=== RUN   TestAccHerokuPipeline_NoOwner
--- PASS: TestAccHerokuPipeline_NoOwner (1.47s)
=== RUN   TestAccHerokuPipeline_InvalidOwnerID
--- PASS: TestAccHerokuPipeline_InvalidOwnerID (0.30s)
=== RUN   TestAccHerokuPipeline_InvalidOwnerType
--- PASS: TestAccHerokuPipeline_InvalidOwnerType (0.30s)
=== RUN   TestAccHerokuSlug_Basic
--- PASS: TestAccHerokuSlug_Basic (7.31s)
=== RUN   TestAccHerokuSlug_NoFile
--- PASS: TestAccHerokuSlug_NoFile (6.34s)
=== RUN   TestAccHerokuSlug_AllOpts
--- PASS: TestAccHerokuSlug_AllOpts (7.38s)
=== RUN   TestAccHerokuSlug_WithFile
--- PASS: TestAccHerokuSlug_WithFile (9.08s)
=== RUN   TestAccHerokuSlug_WithRemoteFile
--- PASS: TestAccHerokuSlug_WithRemoteFile (7.66s)
=== RUN   TestAccHerokuSlug_WithInsecureRemoteFile
--- PASS: TestAccHerokuSlug_WithInsecureRemoteFile (0.31s)
=== RUN   TestAccHerokuSlug_WithFile_InPrivateSpace
--- PASS: TestAccHerokuSlug_WithFile_InPrivateSpace (500.09s)
=== RUN   TestAccHerokuSpaceAppAccess_Basic
Error:     resource_heroku_space_app_access_test.go:21: Step 1/1 error: Check failed: Check 2/2 error: Error in test check: heroku_space_app_access.foobar: Attribute 'permissions.3695762012' not found
        Test check address "permissions.3695762012" likely indexes into TypeSet
        This is currently not possible in the SDK
--- FAIL: TestAccHerokuSpaceAppAccess_Basic (502.58s)
=== RUN   TestAccHerokuSpaceInboundRuleset_Basic
--- PASS: TestAccHerokuSpaceInboundRuleset_Basic (441.37s)
=== RUN   TestAccHerokuSpace_Basic
--- PASS: TestAccHerokuSpace_Basic (442.01s)
=== RUN   TestAccHerokuSpace_Shield
--- PASS: TestAccHerokuSpace_Shield (472.05s)
=== RUN   TestAccHerokuSpace_IPRange
--- PASS: TestAccHerokuSpace_IPRange (473.67s)
=== RUN   TestAccHerokuSpace_CIDRs
--- PASS: TestAccHerokuSpace_CIDRs (902.24s)
=== RUN   TestAccHerokuVPNConnection_basic
Error:     resource_heroku_space_vpn_connection_test.go:20: Step 1/1 error: Error running apply: exit status 1
        
        Error: Error waiting for VPN to become available: empty non-retryable error received. This is a bug with the Terraform provider and should be reported as a GitHub issue in the provider repository.
        
        
--- FAIL: TestAccHerokuVPNConnection_basic (1000.13s)
=== RUN   TestAccHerokuTeamCollaborator_Org
Error:     resource_heroku_team_collaborator_test.go:22: Step 1/1 error: Check failed: Check 3/3 error: Error in test check: heroku_team_collaborator.foobar-collaborator: Attribute 'permissions.1056122515' not found
        Test check address "permissions.1056122515" likely indexes into TypeSet
        This is currently not possible in the SDK
--- FAIL: TestAccHerokuTeamCollaborator_Org (7.09s)
=== RUN   TestAccHerokuTeamCollaboratorPermsOutOfOrder_Org
Error:     resource_heroku_team_collaborator_test.go:49: Step 1/1 error: Check failed: Check 3/3 error: Error in test check: heroku_team_collaborator.foobar-collaborator: Attribute 'permissions.1056122515' not found
        Test check address "permissions.1056122515" likely indexes into TypeSet
        This is currently not possible in the SDK
--- FAIL: TestAccHerokuTeamCollaboratorPermsOutOfOrder_Org (6.98s)
=== RUN   TestAccHerokuTeamMember_Org
--- PASS: TestAccHerokuTeamMember_Org (1.70s)
=== RUN   TestValidateUUID
--- PASS: TestValidateUUID (0.00s)
FAIL
FAIL	github.com/heroku/terraform-provider-heroku/v4/heroku	8267.164s
FAIL
Error: GNUmakefile:21: recipe for target 'testacc' failed
make: *** [testacc] Error 1
Error: Process completed with exit code 2.
```

File: heroku/provider_test.go [hunk 1]
```diff
@@ -9,35 +9,34 @@ import (
...
 
 func TestProvider(t *testing.T) {
-	if err := Provider().(*schema.Provider).InternalValidate(); err != nil {
+	if err := Provider().InternalValidate(); err != nil {
 		t.Fatalf("err: %s", err)
 	}
 }
 
 func TestProvider_impl(t *testing.T) {
-	var _ terraform.ResourceProvider = Provider()
```
Comment (bernerdschaefer) [inline heroku/provider_test.go:36]: This worked to assert that the Provider correctly implemented the ResourceProvider interface, and would generate a compile time error if it didn't. Since the provider is now a concrete type, I'd say we should just delete the test.

File: heroku/resource_heroku_app.go [hunk 2]
```diff
@@ -344,15 +344,15 @@ func setTeamDetails(d *schema.ResourceData, app *application) (err error) {
...
 func setAppDetails(d *schema.ResourceData, app *application) (err error) {
-	err = d.Set("name", app.App.Name)
-	err = d.Set("stack", app.App.Stack)
-	err = d.Set("internal_routing", app.App.InternalRouting)
-	err = d.Set("region", app.App.Region)
-	err = d.Set("git_url", app.App.GitURL)
-	err = d.Set("web_url", app.App.WebURL)
-	err = d.Set("acm", app.App.Acm)
-	err = d.Set("uuid", app.App.ID)
-	err = d.Set("heroku_hostname", fmt.Sprintf("%s.herokuapp.com", app.App.Name))
```
Comment (bernerdschaefer) [inline heroku/resource_heroku_app.go:355]: What's the motivation for this change? It looks like the Set method still returns an error in the v2 SDK.

File: heroku/resource_heroku_build.go [hunk 3]
```diff
@@ -64,10 +64,10 @@ func resourceHerokuBuild() *schema.Resource {
 			},
 
 			"source": {
-				Type:         schema.TypeMap,
-				Required:     true,
-				ForceNew:     true,
-				ValidateFunc: validateSourceUrl,
+				Type:     schema.TypeList,
```
Comment (bernerdschaefer) [inline heroku/resource_heroku_build.go:67]: What's the driver for this type change? Is it backwards compatible?

File: heroku/resource_heroku_team_collaborator.go [hunk 4]
```diff
@@ -134,9 +134,6 @@ func resourceHerokuTeamCollaboratorRead(d *schema.ResourceData, meta interface{}
 }
 
 func resourceHerokuTeamCollaboratorUpdate(d *schema.ResourceData, meta interface{}) error {
-	// Enable Partial state mode to track what was successfully committed
-	d.Partial(true)
```
Comment (bernerdschaefer) [inline heroku/resource_heroku_team_collaborator.go:138]: Why don't we need this any more?

**Review by bernerdschaefer — COMMENTED** [review]
Overall looks good, just have a few questions.

File: heroku/resource_heroku_app.go (hunk 2 above)
Comment (davidji99) [reply heroku/resource_heroku_app.go:355]: Ah, let me undo this change.

File: heroku/resource_heroku_build.go (hunk 3 above)
Comment (davidji99) [reply heroku/resource_heroku_build.go:67]: The v2 sdk does not allow for `Elem` of type `&schema.Resource` for `TypeMap`. You need to switch to using `TypeList`. It is not backwards compatible in that users will need to update their configuration from

```
source = {
   ....
}
```

to

```
source {
}
```

File: heroku/resource_heroku_team_collaborator.go (hunk 4 above)
Comment (davidji99) [reply heroku/resource_heroku_team_collaborator.go:138]:
```
// Partial is a legacy function that was used for capturing state of specific
// attributes if an update only partially worked. Enabling this flag without
// setting any specific keys with the now removed SetPartial has a useful side
// effect of preserving all of the resource's previous state. Although confusing,
// it has been discovered that during an update when an error is returned, the
// proposed config is set into state, even without any calls to d.Set.
```

By my understanding,`d.Partial` is no longer useful here.

File: heroku/provider_test.go (hunk 1 above)
Comment (davidji99) [reply heroku/provider_test.go:36]: Sounds good.

**Review by bernerdschaefer — APPROVED** [review]
